subiquity/server/controllers/filesystem.py

```diff
@@ -223,11 +223,13 @@ def guided(self, choice: GuidedChoiceV2):
             new_size = align_up(choice.target.new_size, part_align)
             if new_size > partition.size:
                 raise Exception(f'Aligned requested size {new_size} too large')
-            gap_offset = partition.offset + new_size
             partition.size = new_size
             partition.resize = True
             mode = 'use_gap'
-            target = gaps.at_offset(disk, gap_offset)
+            target = gaps.after(disk, partition.offset)
```

Comment:
I think a comment here about why you are doing this would be a good idea.

.... although does this mean we are creating an ESP that is a logical partition? I think that doesn't work with our efi tools, unfortunately.

Reply:
https://bugs.launchpad.net/ubuntu/+source/efivar/+bug/1796260

Reply:
I want boot.py to figure that out.  By the time we are in `guided()` we are actually changing the partitions in the model.  We should be able to say in boot.py that it isn't appropriate to create EFI partitions in logical and that there is a primary partition available.  So if we would be forced into such things, I want boot plan to have `can_be_boot_device()` return false, then that scenario isn't even offered as one of the guided possibilities.

I think this PR doesn't make this problem worse, and fixing extended and logical is next on my list, so I'd like to handle it with that work.  What do you think?

Reply:
Yes that makes sense